lib/record_store/provider/google_cloud_dns.rb

```diff
@@ -0,0 +1,121 @@
+require 'google/cloud/dns'
+
+module RecordStore
+  class Provider::GoogleCloudDNS < Provider
+    class << self
+      def apply_changeset(changeset, stdout = $stdout)
+        zone = session.zone(convert_to_name(changeset.zone))
+
+        deletions = convert_records_to_gcloud_record_sets(zone, changeset.current_records)
+        additions = convert_records_to_gcloud_record_sets(zone, changeset.desired_records)
+
+        # The Google API library will handle applying the changeset transactioanlly
+        zone.update(additions, deletions)
+      end
+
+      # returns an array of Record objects that match the records which exist in the provider
+      def retrieve_current_records(zone:, stdout: $stdout)
+        gcloud_record_sets = session.zone(convert_to_name(zone)).records
+
+        records = gcloud_record_sets.map do |record_set|
+          next if record_set.type == 'SOA'
+
+          # Unroll each record set into multiple records
+          record_set.data.map do |record|
+            begin
+              record_set_member = record_set.dup
+              record_set_member.data = [record]
+              build_from_api(record_set_member)
+            rescue StandardError => e
+              stdout.puts "Cannot build record: #{record}"
+            end
+          end
+        end
+
+        # We need to filter out for nil records (i.e. since skip the SOA record)
+        records.flatten.select(&:present?)
```

Comment:
Hmm... if you're trying to just filter out the non-`nil` elements, you can also chain `compact!` to the block. 

This totally works too!

Reply:
TIL 😻